pkg/util/detector/detector.go

```diff
@@ -343,26 +348,58 @@ func (d *ResourceDetector) ApplyClusterPolicy(object *unstructured.Unstructured,
 		return err
 	}
 
-	binding := d.BuildClusterResourceBinding(object, objectKey, policy)
-	bindingCopy := binding.DeepCopy()
-	operationResult, err := controllerutil.CreateOrUpdate(context.TODO(), d.Client, bindingCopy, func() error {
-		// Just update necessary fields, especially avoid modifying Spec.Clusters which is scheduling result, if already exists.
-		bindingCopy.Labels = binding.Labels
-		bindingCopy.OwnerReferences = binding.OwnerReferences
-		bindingCopy.Spec.Resource = binding.Spec.Resource
-		return nil
-	})
-	if err != nil {
-		klog.Errorf("Failed to apply cluster policy(%s) for object: %s. error: %v", policy.Name, objectKey, err)
-		return err
+	labels := map[string]string{
+		util.ClusterPropagationPolicyLabel: policy.GetName(),
 	}
 
-	if operationResult == controllerutil.OperationResultCreated {
-		klog.Infof("Create ClusterResourceBinding(%s) successfully.", binding.GetName())
-	} else if operationResult == controllerutil.OperationResultUpdated {
-		klog.Infof("Update ClusterResourceBinding(%s) successfully.", binding.GetName())
+	// Namespace scoped resoource will lead to a ResourceBinding; cluster scoped resource will lead to a ClusterResourceBinding
```

Comment:
How about giving it more detailed comments, like:
```
	// Build `ResourceBinding` or `ClusterResourceBinding` according to the resource template's scope.
	// For namespace-scoped resources, which namespace is not empty, building `ResourceBinding`.
	// For cluster-scoped resources, which namespace is empty, building `ClusterResourceBinding`.
```

Reply:
Done